src/main/java/com/curity/mongodb/datasource/ConnectionPool.java

```diff
@@ -0,0 +1,46 @@
+package com.curity.mongodb.datasource;
+
+import com.curity.mongodb.datasource.config.MongoDataAccessProviderConfiguration;
+import com.mongodb.MongoClient;
+import com.mongodb.MongoClientOptions;
+import com.mongodb.MongoCredential;
+import com.mongodb.ServerAddress;
+import com.mongodb.client.MongoDatabase;
+import org.bson.codecs.configuration.CodecRegistry;
+import org.bson.codecs.pojo.PojoCodecProvider;
+import se.curity.identityserver.sdk.plugin.ManagedObject;
+
+import static org.bson.codecs.configuration.CodecRegistries.fromProviders;
+import static org.bson.codecs.configuration.CodecRegistries.fromRegistries;
+
+public class ConnectionPool extends ManagedObject<MongoDataAccessProviderConfiguration>
+{
+    private final MongoClient _mongoClient;
+    private final MongoDataAccessProviderConfiguration _configuration;
+
+
+    public ConnectionPool(MongoDataAccessProviderConfiguration configuration)
+    {
+        super(configuration);
+        _configuration = configuration;
+        ServerAddress serverAddress = new ServerAddress(configuration.getHost(), configuration.getPort());
```

Comment:
This is fine for development, but for prod it must be possible to configure the connection with an SSL server truststore.

Reply:
okay, will add todo and handle later.